packages/job-manager/src/batch/batch.spec.ts

```diff
@@ -65,191 +83,97 @@ describe(Batch, () => {
-                .returns(async () => Promise.resolve({ value: [] } as BatchServiceModels.TaskAddCollectionResponse));
-
-            const jobTask = await batch.createTasks(jobId1, messages);
-            cloudTaskListResultFirst[0].id = jobTask[0].id;
-            cloudTaskListResultNext[0].id = jobTask[1].id;
-            jobTasksExpected[0].id = jobTask[0].id;
-            jobTasksExpected[1].id = jobTask[1].id;
-
-            const tasksActual = await batch.getCreatedTasksState(jobId1);
+    describe('createTasks()', () => {
+        it('create no new tasks when no messages provided', async () => {
+            const messages: Message[] = [];
+            const tasksActual = await batch.createTasks(jobId1, messages);
 
-            expect(tasksActual).toEqual(jobTasksExpected);
-            taskMock.verifyAll();
+            expect(tasksActual.length).toBe(0);
         });
 
-        it('verifies taskAddParameter data', async () => {
-            const cloudTaskListResultFirst = <BatchServiceModels.CloudTaskListResult>[];
-            cloudTaskListResultFirst.push({
-                id: jobId1,
-                state: JobTaskState.completed,
-                executionInfo: <BatchServiceModels.TaskExecutionInformation>{
-                    result: JobTaskExecutionResult.success,
-                },
-            });
+        it('should add no more than 100 tasks in a single Batch API call', async () => {
+            const messagesCount = 150;
+            const messages = [];
+            let taskAddCollectionResponse: BatchServiceModels.TaskAddCollectionResponse;
+            const tasksAddedBatchCount: number[] = [];
 
-            const messages = [
-                {
-                    messageId: 'messageId-1',
-                    messageText: JSON.stringify({ msg: 'message 1 text' }),
-                },
-                {
-                    messageId: 'messageId-2',
-                    messageText: JSON.stringify({ msg: 'message 2 text' }),
-                },
-            ];
-            const message1CommandLine = 'message 1 command line';
-            const message2CommandLine = 'message 2 command line';
-
-            runnerTaskConfigMock
-                .setup(t => t.getCommandLine(JSON.parse(messages[0].messageText)))
-                .returns(() => message1CommandLine)
-                .verifiable();
-            runnerTaskConfigMock
-                .setup(t => t.getCommandLine(JSON.parse(messages[1].messageText)))
-                .returns(() => message2CommandLine)
-                .verifiable();
+            for (let i = 0; i < messagesCount; i += 1) {
+                messages.push({
+                    messageText: '{}',
+                    messageId: `message-id-${i}`,
+                });
+            }
 
-            let actualTaskAddParameters: BatchServiceModels.TaskAddParameter[];
             taskMock
                 .setup(async o => o.addCollection(jobId1, It.isAny()))
```

Comment:
> It.isAny() [](start = 57, length = 11)

you should validate the count in each call

Reply:
won't fix validated as result of line 161 tasksAddedBatchCount.push(taskAddParameters.length);